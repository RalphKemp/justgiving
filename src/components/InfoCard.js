import React from 'react';
import styled from 'styled-components';
import { media } from '../helpers/sizing';
import MyCodeLoader from '../helpers/contentLoader';

const InfoCardContainer = styled.div`
  position: relative;
  background-color: #99e0bd;
  height: auto;
  border-radius: 5px;
  margin-top: 10px;
  margin-bottom: 55px;
  font-family: 'Roboto', sans-serif;
  font-weight: 400;
  font-size: 13px;
  padding: 10px;
  ${media.tablet`
      width: 65%;
  `};
`;

const NameDiv = styled.div`
  position: absolute;
  top: -50px;
  left: 60px;
  font-family: 'Roboto', sans-serif;
  font-weight: 500;
  font-size: 23px;
  margin-top: 18px;
  ${media.mid`
    left: 0px;
  `};
`;

const InfoBox = styled.div`
  background-color: white;
  border-radius: 5px;
  padding: 5px;
  margin: 5px;
  width: fit-content;
`;

const CharityInfo = styled.div`
  display: flex;
  flex-direction: column !important;
`;

const SiteLinks = styled.div`

`;

const InfoCard = ({ charityInfo }) => {
  return (
    <InfoCardContainer>
      <div>
        <NameDiv>
          <p>{charityInfo.name}</p>
        </NameDiv>
        <CharityInfo>
          <InfoBox>{charityInfo.description}</InfoBox>
          <InfoBox>{charityInfo.logoAbsoluteUrl}</InfoBox>
          <InfoBox>{charityInfo.thankyouMessage}</InfoBox>
          <SiteLinks>
            <InfoBox><a href={charityInfo.profilePageUrl}>{charityInfo.profilePageUrl}</a></InfoBox>
            <InfoBox><a href={charityInfo.websiteUrl}>{charityInfo.websiteUrl}</a></InfoBox>
          </SiteLinks>
          <InfoBox><a href={"mailto:"+charityInfo.emailAddress}>{charityInfo.emailAddress}</a></InfoBox>
        </CharityInfo>
      </div>
    </InfoCardContainer>
  );
};

export default InfoCard;
