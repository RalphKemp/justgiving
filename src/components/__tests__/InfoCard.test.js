import React from 'react';
import { shallow } from 'enzyme';
// using curly braces to import component without state.
import { InfoCard } from '../InfoCard.js';

describe('InfoCard', () => {

  const defaultProps = {
    name: 'charity',
    logoAbsoluteUrl: 'url',
    description: 'this is the desc',
    thankyouMessage: 'hello',
    profilePageUrl: 'url2',
    websiteUrl: 'url3',
    emailAddress: '@@'
  }

  it('renders correctly', () => {
    const wrapper = shallow(<InfoCard charityInfo={defaultProps}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
