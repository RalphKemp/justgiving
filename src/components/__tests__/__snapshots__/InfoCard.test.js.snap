// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard renders correctly 1`] = `
<styled.div>
  <div>
    <styled.div>
      <p>
        charity
      </p>
    </styled.div>
    <styled.div>
      <styled.div>
        <img
          src="url"
        />
        <Styled(styled.div)>
          this is the desc
        </Styled(styled.div)>
      </styled.div>
      <styled.div>
        hello
      </styled.div>
      <styled.div>
        <Styled(styled.div)
          purple={true}
        >
          <a
            href="url2"
          >
            View charity page
          </a>
        </Styled(styled.div)>
        <Styled(styled.div)>
          <a
            href="url3"
          >
            Visit site
          </a>
        </Styled(styled.div)>
        <Styled(styled.div)>
          <a
            href="mailto:@@"
          >
            Contact
          </a>
        </Styled(styled.div)>
      </styled.div>
    </styled.div>
  </div>
</styled.div>
`;
